config/envoyconfig/listeners.go

```diff
@@ -574,6 +568,39 @@ func (b *Builder) buildDownstreamValidationContext(
 	return vc
 }
 
+// clientCABundle returns a bundle of the globally configured client CA and any
+// per-route client CAs.
+func clientCABundle(ctx context.Context, cfg *config.Config) []byte {
+	var bundle bytes.Buffer
+	ca, _ := cfg.Options.GetClientCA()
+	addCAToBundle(&bundle, ca)
+	allPolicies := cfg.Options.GetAllPolicies()
+	for i := range allPolicies {
+		p := &allPolicies[i]
+		if p.TLSDownstreamClientCA == "" {
```

Comment:
required: I think we need to account for `TLSDownstreamClientCAFile` too.

I *think* we handle file changes automatically in the config layer, so it's probably sufficient just to reproduce similar logic as `GetClientCA`

Reply:
Denis had the same question on the previous v0.17 PR: https://github.com/pomerium/pomerium/pull/4228#discussion_r1222122189.

I agree it looks weird, but I think we don't need to check `TLSDownstreamClientCAFile` because the Policy `Validate()` method will populate `TLSDownstreamClientCA` when a file is configured. See config/policy.go here: https://github.com/pomerium/pomerium/blob/26bbcdfe07a81204b4357b15998e2fe152f7669f/config/policy.go#L506-L512

I've added a comment to help explain this. I will also test this locally.

Reply:
Setting `tls_downstream_client_ca_file` works for me locally when building with these changes.